spotify_tensorflow/dataset.py

```diff
@@ -138,45 +119,28 @@ def _try_enable_sharding(dataset):
         return dataset
 
     @classmethod
-    def mk_training_iter(cls, training_data_dir, feature_mapping_fn):
+    def mk_iter(cls, data_dir,
+                scope="tfrecords_iter",
+                feature_mapping_fn=None):
         """Make a training `Dataset` iterator.
 
         Args:
-            training_data_dir: a directory contains training data.
-            feature_mapping_fn: A function which maps `FeatureInfo` to `FixedLenFeature` or
-                `VarLenFeature` values. Default maps all features to
-                tf.FixedLenFeature((), feature_info.type, default_value=0).
-
-        Returns:
-            A `Dataset` that should be used for training purposes.
-        """
-        with tf.name_scope("training-input"):
-            train_dataset, _ = cls._get_featran_example_dataset(
-                training_data_dir,
-                feature_mapping_fn=feature_mapping_fn)
-            return cls._mk_iterator(train_dataset)
-
-    @classmethod
-    def mk_eval_iter(cls, eval_data_dir, feature_mapping_fn):
-        """Make an evaluation `Dataset` iterator.
-
-        Args:
-            eval_data_dir: a directory contains evaluation data.
+            data_dir: a directory contains training data.
+            scope: TF name score for this op.
             feature_mapping_fn: A function which maps `FeatureInfo` to `FixedLenFeature` or
                 `VarLenFeature` values. Default maps all features to
                 tf.FixedLenFeature((), feature_info.type, default_value=0).
 
         Returns:
-            A `Dataset` that should be used for evaluation purposes.
+            A `Dataset` that should be used for training purposes and a `DatasetContext` object.
```

Comment:
Should probably return `iterator` not `Dataset` right? Also I still kinda don't like the fact that it's hard coded `one_shot_iterator`, should we add a parameter to the method for func `Dataset => Iterator`, with default value `_mk_one_shot_iterator`?